thoth/storages/graph/janusgraph.py

```diff
@@ -626,7 +622,36 @@ def sync_solver_result(self, document: dict) -> None:
             except Exception:  # pylint: disable=broad-except
                 _LOGGER.exception("Failed to sync unsolvable Python package, error is not fatal: %r", unsolvable)
 
-    def _deb_sync_analysis_result(self, document: dict, runtime_environment: RuntimeEnvironment) -> None:
+        for unparsed in document['result']['unparsed']:
+            parts = unparsed.rsplit('==', maxsplit=1)
+            if len(parts) != 2:
+                # This request did not come from graph-refresh job as there is not pinned version.
+                _LOGGER.warning(
+                    f"Cannot sync unparsed package {unparsed} as package is not locked to as specific version"
+                )
+                continue
+
+            package_name, package_version = parts
+            try:
+                python_package, _, python_package_version = self.create_pypi_package_version(
+                    package_name=package_name,
+                    package_version=package_version,
+                )
+
+                Solved.from_properties(
+                    source=ecosystem_solver,
+                    target=python_package_version,
+                    solver_document_id=solver_document_id,
+                    solver_datetime=solver_datetime,
+                    solver_error=True,
+                    solver_error_unsolvable=False,
+                    solver_error_unparsable=True
+                ).get_or_create(self.g)
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Failed to sync unparsed Python package, error is not fatal: %r", unparsed)
+
+    def _deb_sync_analysis_result(self, document_id: str, document: dict,
+                                  runtime_environment: RuntimeEnvironment) -> None:
```

Comment:
This is from https://github.com/thoth-station/storages/pull/212, it has to go in first due to merge conflicts.